### ISSUE TITLE
high severity in jsonwebtoken  <=8.5.1

### ISSUE THREAD

**Issue by huineng** [box/box-node-sdk#793]
I see that a pull was made , but closed https://github.com/box/box-node-sdk/pull/787  

I  think there's not much choice then to bump this version

thanks

for info the full track 

```
jsonwebtoken  <=8.5.1
Severity: high
jsonwebtoken unrestricted key type could lead to legacy keys usage  - https://github.com/advisories/GHSA-8cf7-32gw-wr33
jsonwebtoken has insecure input validation in jwt.verify function - https://github.com/advisories/GHSA-27h2-hvpr-p74q
jsonwebtoken's insecure implementation of key retrieval function could lead to Forgeable Public/Private Tokens from RSA to HMAC - https://github.com/advisories/GHSA-hjrf-2m68-5959
jsonwebtoken vulnerable to signature validation bypass due to insecure default algorithm in jwt.verify() - https://github.com/advisories/GHSA-qwph-4952-7xr6
No fix available
node_modules/box-node-sdk/node_modules/jsonwebtoken
  box-node-sdk  *
  Depends on vulnerable versions of jsonwebtoken
  node_modules/box-node-sdk
```

**Comment by antusus** [comment]
Hi,
You can check in my comment that in version 9 of Jsonwebtoken they removed support for Node versions 11 and below. We wold need to drop support as well in Node SDK which is a breaking change and will take some time to properly communicate to external customers.

**Comment by shahrin014** [comment]
I think it is a natural expectation to have to upgrade Node in order to have access to updated library versions. For our organization the `jsonwebtoken` version is a big issue, and we are monitoring this thread closely.

Note: I am not related to @huineng, but also face the same situation in regards to the `jsonwebtoken` library.

**Comment by huineng** [comment]
they have been downgraded to moderate

``` 
jsonwebtoken  <=8.5.1
Severity: moderate
jsonwebtoken unrestricted key type could lead to legacy keys usage  - https://github.com/advisories/GHSA-8cf7-32gw-wr33
jsonwebtoken vulnerable to signature validation bypass due to insecure default algorithm in jwt.verify() - https://github.com/advisories/GHSA-qwph-4952-7xr6
jsonwebtoken's insecure implementation of key retrieval function could lead to Forgeable Public/Private Tokens from RSA to HMAC - https://github.com/advisories/GHSA-hjrf-2m68-5959
```

**Comment by lukaszsocha2** [comment]
Hi,
we are already working on new 3.0 Node SDK version, which will drop deprecated Node versions and upgrade all dependencies with issues. The release is planned somewhere between May and July 2023.
Best,
@lukaszsocha2

**Comment by huineng** [comment]
thanks,  meanwhile , we did an override in our package json and still with v 9 everything seems to work, but at least our vulnerabilities are gone

```
***
"overrides": {
        "box-node-sdk": {
            "jsonwebtoken": "9.0.0",
            "@types/jsonwebtoken": "9.0.0"
        }
    },
***

**Comment by congminh1254** [comment]
Hi @huineng 
We have just released version 3.0 of the Box Node SDK. You can bump the version from now, version 2.x will be maintained until July 23, 2023.
I'm closing this thread for now.
Best regards,
Minh
